src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,220 @@
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, id);
+            return statementForCars.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "select cars.id as c_id, cars.model "
+                + " as c_m, drivers.id as d_id, drivers.name d_n,"
+                + " drivers.license_number as d_ln, manufacturers.id "
+                + " as m_id, manufacturers.name as m_n,"
+                + " manufacturers.country as m_c from cars"
+                + " inner join cars_drivers on cars.id = cars_drivers.car_id"
+                + " inner join drivers on drivers.id = cars_drivers.driver_id"
+                + " inner join manufacturers on manufacturers.id = cars.manufacturer_id"
+                + " where drivers.id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCarsFromDB(resultSet));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private Car getCarFromDB(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        Manufacturer manufacturer = new Manufacturer();
+        resultSet.beforeFirst();
+        if (resultSet.next()) {
+            manufacturer.setId(resultSet.getObject("c_mid", Long.class));
+            manufacturer.setName(resultSet.getObject("m_n", String.class));
+            manufacturer.setCountry(resultSet.getObject("m_c", String.class));
+            car.setId(resultSet.getObject("c_id", Long.class));
+            car.setModel(resultSet.getObject("c_md", String.class));
+            car.setManufacturer(manufacturer);
+        }
+        while (resultSet.next()) {
+            Driver driver = new Driver();
+            driver.setId(resultSet.getObject("d_id", Long.class));
+            driver.setName(resultSet.getObject("d_n", String.class));
+            driver.setLicenceNumber(resultSet.getObject("d_ln", String.class));
+            car.setDrivers(driver);
+        }
+        return car;
```

Comment:
```suggestion
    private Car getCarFromDB(ResultSet resultSet) throws SQLException {
        Car car = new Car();
        Manufacturer manufacturer = new Manufacturer();
        manufacturer.setId(resultSet.getObject("c_mid", Long.class));
        manufacturer.setName(resultSet.getObject("m_n", String.class));
        manufacturer.setCountry(resultSet.getObject("m_c", String.class));
        car.setId(resultSet.getObject("c_id", Long.class));
        car.setModel(resultSet.getObject("c_md", String.class));
        car.setManufacturer(manufacturer);
        return car;
```
let's fetch the info about drivers in a separate query in a separate method